detox/src/devices/drivers/android/AndroidDriver.js

```diff
@@ -71,8 +74,16 @@ class AndroidDriver extends DeviceDriverBase {
 
   async uninstallApp(deviceId, bundleId) {
     await this.emitter.emit('beforeUninstallApp', { deviceId, bundleId });
-    const uninstallHelper = new AppUninstallHelper(this.adb);
-    await uninstallHelper.uninstall(deviceId, bundleId);
+    await this.appUninstallHelper.uninstall(deviceId, bundleId);
+  }
+
+  async installUtilBinaries(deviceId, paths) {
+    for (const path of paths) {
+      const packageId = await this.getBundleIdFromBinary(path);
+      if (!await this.adb.isPackageInstalled(deviceId, packageId)) {
+        await this.appInstallHelper.install(deviceId, path);
```

Comment:
What if the device has older version installed? Should we overwrite, hence update it?

Reply:
See my answer to the previous comment... also, I guess I'm allergic to single functions taking many arguments which result in various `if`'s, etc.

Reply:
As per version overwrite, I admit I'm not entirely comfortable about it as well, but this isn't a real-life use case ATM. I will try to scratch my head again and see whether I could make it feasible nonetheless.

Reply:
👍  @d4vidi , probably you are right here. Just trying to wrap my head around how we publish this API to the general public... 🤔  The current implementation level feels like protected/internal, and we are adding one more thing to the public Device class and its configuration, which by this PR design is not intended to work on both platforms. 🤯 I still feel like we might add some tech debt by this, this is legit, but eventually we should think on total rewriting Device class.

Reply:
I thought this over and realized that in order to properly solve this, we need to have global init & teardown. Only then, will I be able to make sure that the APK is installed _once_ - overwriting any existing version, and then refrain from doing so in each suite.

I'll work out an issue explaining what needs to be done and what is blocked by it (this isn't the only thing, of course).

Reply:
As per platform, well - there's technically no one to take this on iOS. I'm as unsatisfied as you are.